docs/how-to/hip_runtime_api/multi_device.rst

```diff
@@ -0,0 +1,423 @@
+        hipEventElapsedTime(&milliseconds0, startEvent0, stopEvent0);
+        hipEventElapsedTime(&milliseconds1, startEvent1, stopEvent1);
+
+        std::cout << "Elapsed time on GPU 0: " << milliseconds0 << " ms" << std::endl;
+        std::cout << "Elapsed time on GPU 1: " << milliseconds1 << " ms" << std::endl;
+
+        // Cleanup for device 0
+        hipSetDevice(0);
+        hipEventDestroy(startEvent0);
+        hipEventDestroy(stopEvent0);
+        hipStreamSynchronize(stream0);
+        hipStreamDestroy(stream0);
+        hipFree(deviceData0);
+
+        // Cleanup for device 1
+        hipSetDevice(1);
+        hipEventDestroy(startEvent1);
+        hipEventDestroy(stopEvent1);
+        hipStreamSynchronize(stream1);
+        hipStreamDestroy(stream1);
+        hipFree(deviceData1);
+
+        return 0;
+    }
+
+Peer-to-peer memory access
+===============================================================================
+
+In multi-GPU systems, peer-to-peer memory access enables one GPU to directly
+read or write to the memory of another GPU. This capability reduces data
+transfer times by allowing GPUs to communicate directly without involving the
+host. Enabling peer-to-peer access can significantly improve the performance of
+applications that require frequent data exchange between GPUs, as it eliminates
+the need to transfer data through the host memory.
+
+By adding peer-to-peer access to the example referenced in
+:ref:`multi_device_selection`, data can be copied between devices:
+
+.. tab-set::
+
+    .. tab-item:: with peer-to-peer
+
+        .. code-block:: cpp
+            :emphasize-lines: 31-36, 50-54
+
+            #include <hip/hip_runtime.h>
+            #include <iostream>
+
+            #define HIP_CHECK(expression)                        \
+            {                                                    \
+                const hipError_t status = expression;            \
+                if (status != hipSuccess) {                      \
+                    std::cerr << "HIP error " << status          \
+                            << ": " << hipGetErrorString(status) \
+                            << " at " << __FILE__ << ":"         \
+                            << __LINE__ << std::endl;            \
+                    exit(status);                                \
+                }                                                \
+            }
+
+            __global__ void simpleKernel(double *data)
+            {
+                int idx   = blockIdx.x * blockDim.x + threadIdx.x;
+                data[idx] = idx * 2.0;
+            }
+
+            int main()
+            {
+                double* deviceData0;
+                double* deviceData1;
+                size_t  size = 1024 * sizeof(*deviceData0);
+
+                int deviceId0 = 0;
+                int deviceId1 = 1;
+
+                // Enable peer access
+                hipSetDevice(deviceId0);
```

Comment:
I would note in the comment that the device must be the active device before enabling peer access to another device. Also, shouldn't these lines come later, after the memory allocation for Device 0 and 1 are allocated?

Reply:
Thank! It's a good idea. I extended the comment.